frappe/core/doctype/role_profile/role_profile.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+
+class RoleProfile(Document):
+	def autoname(self):
+		"""set name as Role Name"""
+		self.name = self.role_name
+
+	def on_update(self):
+		roles = [x.role for x in self.roles]
+		users = frappe.get_all('User', filters={'role_name': self.role_name})
+
+		for d in users:
```

Comment:
this should in the `user.py` controller. So when user is saved, the roles must also be update. You only call the method here from the user controller